### PR TITLE
ATT-145: SuperAdmin Invites User and Assign Permissions

### PR DESCRIPTION
**Pull request by altairrostam**

### CODE REVIEW

File: app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+import Button from "@mui/material/Button";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import Grid from "@mui/material/Grid";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ListItemText from "@mui/material/ListItemText";
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx:11]: Nice

File: app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        setRight(not(right, rightChecked));
+        setChecked(not(checked, rightChecked));
+    };
+
+    const handleAllLeft = () => {
+        setLeft(left.concat(right));
+        setRight([]);
+    };
+    const customList = (items) => (
+        <Paper sx={{ width: 200, height: 230, overflow: "auto" }}>
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx:83]: For the height width, use the useTheme hook the calculate the appropriate space: i.e

`
import {useTheme} from "@mui/material"
const theme = useTheme()

height=theme.spacing(10) -> 160px

`

File: app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    return (
+        <>
+            <DialogTitle
+                sx={{
+                    textAlign: "center",
+                }}
+            >
+                <Title>Assign Organizations</Title>
+            </DialogTitle>
+            <DialogContent sx={{ p: 5, mt: 3 }}>
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx:129]: use mui spacing here too

File: app/(protected)/dashboard/(admin)/user/accounts/form/AssignPermissions.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+import FormGroup from "@mui/material/FormGroup";
+import FormControlLabel from "@mui/material/FormControlLabel";
+import Checkbox from "@mui/material/Checkbox";
+import { useInviteUser } from "aria-api/users/users";
+import { Permissions } from "common/enum";
+
+const Title = styled(Typography)(({ theme }) => ({
+    color: theme.palette.primary.main,
+    fontSize: theme.spacing(2.5),
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/form/AssignPermissions.tsx:25]: nice

File: app/(protected)/dashboard/(admin)/user/accounts/table.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import Typography from "@mui/material/Typography";
+import { styled } from "@mui/material/styles";
+
+import CheckIcon from "@mui/icons-material/Check";
+import HighlightOffIcon from "@mui/icons-material/HighlightOff";
+
+const HeaderTypography = styled(Typography)(({ theme }) => ({
+    color: "#100E6A",
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/table.tsx:21]: perhaps somewhere in the theme?

File: app/(protected)/dashboard/(admin)/user/accounts/table.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        "Job",
+        "Department",
+        "Permissions",
+        "Email",
+        "Status",
+        "Action",
+    ];
+
+    useEffect(() => {
+        console.log(data);
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/table.tsx:45]: remove console.log here

File: app/(protected)/dashboard/(admin)/user/accounts/table.tsx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        if (data?.data?.user_organization_requests) {
+            setDataArr(data.data.user_organization_requests);
+        }
+    }, [data]);
+
+    return (
+        <>
+            <TableContainer component={"div"}>
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/table.tsx:53]: I will advice we use React table for tables -> https://react-table-v7.tanstack.com/

File: app/(protected)/dashboard/(admin)/user/accounts/table.tsx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+                <Table sx={{ minWidth: 650 }} aria-label="simple table">
```
Comment (kelvinndmo) [inline app/(protected)/dashboard/(admin)/user/accounts/table.tsx:54]: the aria label is really not relevant here,right, based on the value given

**Review by kelvinndmo — APPROVED** [review]

**Review by kelvinndmo — CHANGES_REQUESTED** [review]
A fe things to fix here @altairrostam

File: app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx (hunk 2 above)
Comment (altairrostam) [reply app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx:83]: fixed

File: app/(protected)/dashboard/(admin)/user/accounts/table.tsx (hunk 5 above)
Comment (altairrostam) [reply app/(protected)/dashboard/(admin)/user/accounts/table.tsx:21]: fixed

File: app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx (hunk 3 above)
Comment (altairrostam) [reply app/(protected)/dashboard/(admin)/user/accounts/form/AssignOrganizations.tsx:129]: done

File: app/(protected)/dashboard/(admin)/user/accounts/table.tsx (hunk 7 above)
Comment (altairrostam) [reply app/(protected)/dashboard/(admin)/user/accounts/table.tsx:53]: ok, will refactor table to become a reusable component too

File: components/displays/table/AriaTable.tsx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+import { useTable } from "react-table";
+
+const HeaderTypography = styled(Typography)(({ theme }) => ({
+    color: theme.palette.info.main,
+    fontWeight: 500,
+    fontSize: theme.spacing(1.75),
+}));
+
+const AriaTable = ({ columns, data }) => {
+    const { getTableProps, getTableBodyProps, headerGroups, rows, prepareRow } =
```
Comment (kelvinndmo) [inline components/displays/table/AriaTable.tsx:20]: Good stuff

**Review by kelvinndmo — APPROVED** [review]

**Comment by kelvinndmo** [comment]
Looks good to me, good job @altairrostam

**Pull request merged**